extensions/spatial/src/org/sbml/jsbml/validator/offline/constraints/GeometryConstraints.java

```diff
@@ -0,0 +1,420 @@
+		  		
+		  		func = new ValidationFunction<Geometry>() {
+		  			
+		  			@Override
+		  			public boolean check(ValidationContext ctx, Geometry geom) {
+		  				
+		  				if(geom.isSetListOfDomains()) {
+		  					return new UnknownElementValidationFunction<ListOf<Domain>>().check(ctx, geom.getListOfDomains());
+		  				}
+		  				return true;
+		  			}
+		  		};
+		  		break;
+		  	}
+		  	
+		  	case SPATIAL_23709:
+		  	{
+		  		// Apart from the general notes and annotations subobjects permitted on all SBML objects, a 
+		  		// ListOfAdjacentDomains container object may only contain AdjacentDomains objects.
+		  		
+		  		func = new ValidationFunction<Geometry>() {
+		  			
+		  			@Override
+		  			public boolean check(ValidationContext ctx, Geometry geom) {
+		  				
+		  				if(geom.isSetListOfAdjacentDomains()) {
+		  					return new UnknownElementValidationFunction<ListOf<AdjacentDomains>>().check(ctx, geom.getListOfAdjacentDomains());
+		  				}
+		  				return true;
+		  			}
+		  		};
+		  		break;
+		  	}
+		  	
+		  	case SPATIAL_23710:
+		  	{
+		  		// Apart from the general notes and annotations subobjects permitted on all SBML objects, a 
+		  		// ListOfGeometryDefinitions container object may only contain GeometryDefinition objects.
+		  		
+		  		func = new ValidationFunction<Geometry>() {
+		  			
+		  			@Override
+		  			public boolean check(ValidationContext ctx, Geometry geom) {
+		  				
+		  				if(geom.isSetListOfGeometryDefinitions()) {
+		  					return new UnknownElementValidationFunction<ListOf<GeometryDefinition>>().check(ctx, geom.getListOfGeometryDefinitions());
+		  				}
+		  				return true;
+		  			}
+		  		};
+		  		break;
+		  	}
+		  	
+		  	case SPATIAL_23711:
+		  	{
+		  		// Apart from the general notes and annotations subobjects permitted on all SBML objects, a 
+		  		// ListOfSampledFields container object may only contain SampledField objects. 
+		  		
+		  		func = new ValidationFunction<Geometry>() {
+		  			
+		  			@Override
+		  			public boolean check(ValidationContext ctx, Geometry geom) {
+		  				
+		  				if(geom.isSetListOfSampledFields()) {
+		  					return new UnknownElementValidationFunction<ListOf<SampledField>>().check(ctx, geom.getListOfSampledFields());
+		  				}
+		  				return true;
+		  			}
+		  		};
+		  		break;
+		  	}
+		  	
+		  	case SPATIAL_23712:
+		  	{
+		  		// A ListOfCoordinateComponents object may have the optional SBML Level 3 Core attributes 
+		  		// metaid and sboTerm. No other attributes from the SBML Level 3 Core namespaces are permitted 
+		  		// on a ListOfCoordinateComponents object.
+		  		
+		  		func = new ValidationFunction<Geometry>() {
+		  			
+		  			@Override
+		  			public boolean check(ValidationContext ctx, Geometry geom) {
+		  				
+		  				if(geom.isSetListOfCoordinateComponents()) {
+		  					return new UnknownCoreAttributeValidationFunction<ListOf<CoordinateComponent>>().check(ctx, geom.getListOfCoordinateComponents())
```

Comment:
You can use the generic method there, no need to call both methods. Same for the following constraints.

Reply:
Okay, sure!